### PR TITLE
vertex tool updates in 3.4 and 3.6 #3450

### PR DESCRIPTION
**Pull request by lajunek**

### CODE REVIEW

**Comment by lajunek** [comment]
Ok, see what happens

**Comment by lajunek** [comment]
@DelazJ this is my first git commit/PR can you please review and pull ?

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 1]
```diff
@@ -496,20 +497,16 @@ vertex in the map canvas, and vice versa. Simply change a coordinate in the tabl
...
-  .. figure:: img/vertex_editor_panel_contextual_button.png
-     :align: center
-
-     Button to open the vertex editor panel via right-click
-
-
+.. note:: **Changed behavior in QGIS 3.4.5**
+    
+   In earlier QGIS 3 versions, right click on a feature would bring up a small popup menu
+   that offers display of the vertex editor. Now, right click on a feature will immediately
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]:
```suggestion
   that offered display of the vertex editor. Now, right click on a feature will immediately
```

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 2]
```diff
@@ -496,20 +497,16 @@ vertex in the map canvas, and vice versa. Simply change a coordinate in the tabl
...
+   show the vertex editor and lock this feature thus disabling the editing of any other features.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]:
```suggestion
   show the vertex editor and lock this feature, thus disabling the editing of any other features.
```

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 3]
```diff
@@ -496,20 +497,16 @@ vertex in the map canvas, and vice versa. Simply change a coordinate in the tabl
...
+   While being locked a feature is exclusive for editing: Selecting and moving of vertices and
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]:
```suggestion
   While being locked, a feature is exclusive for editing: Selecting and moving of vertices and
```

Comment (lajunek) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: Hello havatv I'm new to github, does this mean I have to make the changes in the *.rst files or did you already commit the suggested changes?

Comment (havatv) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: Hi @lajunek !
I have not committed.  It is up to you if you would like to accept my suggestions.
You can accept suggestions by using the "Commit suggestion" button that is shown below the suggestion on github.  Once you have done that, the suggestion will be included in your "pull request".
https://help.github.com/en/articles/incorporating-feedback-in-your-pull-request

**Comment by SrNetoChan** [comment]
@lajunek does this has specific things that are 3.6 only?

**Comment by lajunek** [comment]
fix #3450 fix #3482

**Comment by SrNetoChan** [comment]
@lajunek I see that you have closed #3350, in future PRs, can I ask you to do that only when the PR is merged?

**Comment by DelazJ** [comment]
> @lajunek does this has specific things that are 3.6 only?

All the current three versions are equal in this area so no, nothing specific to 3.6
>I see that you have closed #3350, in future PRs

I guess you meant #3450. Actually I don't think that it has to be closed since not all the reported features are documented (cf first msg - misses the way to add new vertex "in-place", which is something really new and strongly asked)
> fix #3450 fix #3482

From my experience (I did not read anything about), this has to be put in the first message of the PR; in that case, merging the PR closes the report.

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 4]
```diff
@@ -496,20 +517,16 @@ vertex in the map canvas, and vice versa. Simply change a coordinate in the tabl
...
-
-  .. figure:: img/vertex_editor_panel_contextual_button.png
-     :align: center
-
-     Button to open the vertex editor panel via right-click
-
-
+.. note:: **Changed behavior in QGIS 3.4**
+    
+   In earlier QGIS 3 versions, right click on a feature would bring up a small popup menu
```
Comment (m-kuhn) [inline source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: Is it important to explain how it was earlier?
IMHO historical considerations should be avoided and only the current way to do things explained.

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 5]
```diff
@@ -476,6 +477,26 @@ Red circles will appear when hovering vertices.
   selection can jump to the nearest vertex or line. You can use Advanced
   Digitizing Panel constraints for distance, angles, exact X Y location
   before the second click.
+  
+  Here you can use the snap-to-grid feature. Having activated the Remove duplicate nodes and set
+  a value for the Geometry precision in the layer properties, a grid appears on a zoom level
+  according to the Geometry precision.
+  
+  
+  .. _figure_select_vertex_snap_to_grid:
```
[the remark this answers is not in the document]
Comment (lajunek) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: @DelazJ  : Commited the merged images!

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst (hunk 4 above)
Comment (lajunek) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: @m-kuhn 
@DelazJ 
@havatv 
I removed the information about earlier QGIS versions

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 6]
```diff
@@ -476,6 +477,26 @@ Red circles will appear when hovering vertices.
   selection can jump to the nearest vertex or line. You can use Advanced
   Digitizing Panel constraints for distance, angles, exact X Y location
   before the second click.
+  
+  Here you can use the snap-to-grid feature. Having activated the Remove duplicate nodes and set
```
[the remark this answers is not in the document]
Comment (lajunek) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: @DelazJ @m-kuhn 
I removed the duplicate nodes mention

[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: "the duplicate nodes mention" does not seem to have been removed.

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 7]
```diff
@@ -476,6 +477,26 @@ Red circles will appear when hovering vertices.
...
+  a value for the Geometry precision in the layer properties, a grid appears on a zoom level
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: Why was this not committed?

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: "the information about earlier QGIS versions" does not seem to have been removed.

**Pull request merged**